File: #2551.put-marbles-in-bags.cpp
class Solution {
public:
    long long putMarbles(vector<int>& weights, int k) {
        
        long long size = weights.size();
        if (k ==1 || k == size) {
            return 0;
        }

        vector<long long> v;
        for(long long i = 0; i < size - 1; i++) {
            v.emplace_back(weights[i] + weights[i + 1]);
        }

        std::sort(v.begin(), v.end());

        long long max = 0;
        long long min = 0;
        for (long long i = 0; i < k - 1; i++) {
            min = min + v[i];
            max = max + v[v.size() - i - 1]; 
        }
        return max-min;
    }
};

File: #2551.put-marbles-in-bags.js
/**
 * @param {number[]} weights
 * @param {number} k
 * @return {number}
 */
var putMarbles = function(weights, k) {
    if (k === 1) {
        return 0;
    }

    let arr = weights.slice(0, -1).map((v, i) => v + weights[i + 1]);
    arr.sort((a, b) => a - b);
    let min = arr.slice(0, k - 1).reduce((sum, v) => sum + v, 0);
    let max = arr.slice(-(k - 1)).reduce((sum, v) => sum + v, 0);
    return max - min;
};
